### ISSUE TITLE
CircleAvatar() place child below image so it failover to child

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#20645]
Place the child widget below the Image property so that it will show the child if the image is not present.

**Comment by goderbauer** [comment]
I don't understand what's requested here. Can you be more specific?

**Comment by lukepighetti** [comment]
If you provide a child and an image the child is on top of the image. In my opinion this is useless.

But, if you put the child BELOW the image (in a stack) then if the image fails to load it will failover to the child. This is useful.

**Comment by Amitpatil215** [comment]
<img src="https://user-images.githubusercontent.com/54329870/101046168-d3d92a00-35a6-11eb-81d9-a1051564366f.png" width="350">

<details>
<summary>Reproducible Sample</summary>

```Dart
class ExampleCirleAvatar extends StatelessWidget {
  static const imageUrl =
      "https://images.pexels.com/photos/5740932/pexels-photo-5740932.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940";
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: CircleAvatar(
          radius: 60.0,
          backgroundImage: NetworkImage(
            imageUrl,
          ),
          child:
              Center(child: Text("Child Text", style: TextStyle(fontSize: 21))),
          onBackgroundImageError: (exception, stackTrace) {
            print('Exception is $exception');
          },
        ),
      ),
    );
  }
}
```
</details>

```Child``` property is typically used as a ```Text("AB")``` for user's initials. Refer [Documentation](https://api.flutter.dev/flutter/material/CircleAvatar-class.html#material.CircleAvatar.2)

**Comment by lukepighetti** [comment]
Nice! Thanks.

**Comment by lukepighetti** [comment]
I see what you're getting at, but CircleAvatar really needs a backgroundChild property so we can have a fallback image of some kind.

**Comment by Amitpatil215** [comment]
@Hixie Gentle Ping

I would like to be assigned to this issue.

**Workaround:**
Adding property  ```ImageProvider?<Object> fallbackImage```. Typically used with ```AssetImage``` for rolling back if ```backgorundImage``` fails.

**Comment by Hixie** [comment]
@Amitpatil215 You don't need to be assigned to an issue to work on it. :-) Just submit a PR. :-)

Do we want a `backgroundImage` or a `backgroundWidget`? Or a `fallbackImage` or `fallbackWidget`? Not sure which is the right solution here exactly. (Background would always be there, fallback would only be there when the given image fails; images would be specifically images, and widgets would be any widgets, which is more flexible but may be harder to optimize...)

**Comment by Amitpatil215** [comment]
@Hixie, 
Thank you, for the feedback! I agree with you :)

> which is more flexible but may be harder to optimize...)

Because of this I stepped back from widget to image. 

Let's say we added ```fallbackWidget```, In this case if user tries to implement ``fallbackImage`` using ```Image.Network()```

<details>
<summary>Code Sample</summary>

```Dart
CircleAvatar(
          radius: 60.0,
		  backgroundImage: NetworkImage(imageUrl_1), //fails cause of bad request
          fallbackWidget: Image.network(imageUrl_2), 
        ),
```


</details>

<img src="https://user-images.githubusercontent.com/54329870/101076718-cd0fde80-35c9-11eb-85a5-7bfcac1fe50b.png" width="350"> 

Clipping the image is the extra effort a user might be doing. We should consider doing all that hassle at our end by implementing ```fallbackImage```. Does this make sense?
